### PR TITLE
chore: make SelectBuilder public

### PR DESCRIPTION
**Pull request by ClSlaid**
Signed-off-by: ClSlaid <cailue@bupt.edu.cn>

I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary
I'm implementing https://github.com/datafuselabs/databend/issues/9094 and noticed this in table.rs, this could make building rewrite query more accurate and easy to read.

- move SelectBuilder in common_sql::planner::binder::ddl::table outside
- add unit tests for SelectBuilder

### CODE REVIEW

File: src/query/sql/src/planner/utils.rs [hunk 1]
```diff
@@ -0,0 +1,80 @@
+// Copyright 2022 Datafuse Labs.
+//
```
Comment (BohuTANG) [inline src/query/sql/src/planner/utils.rs:2]: Avoid using the file name `utils.rs`,  it will be fat and fat. Please choose a meaningful name for the file.

**Comment by BohuTANG** [comment]
The summary did not tell why move out the `SelectBuilder `, can you add more about it?

**Comment by ClSlaid** [comment]
> The summary did not tell why move out the `SelectBuilder `, can you add more about it?

I'm implementing #9094 and noticed this in table.rs, this could make query building more accurate and easy to read.

File: src/query/sql/src/planner/query_builders.rs [hunk 2]
```diff
@@ -0,0 +1,80 @@
+// Copyright 2022 Datafuse Labs.
```
Comment (BohuTANG) [inline src/query/sql/src/planner/query_builders.rs:1]: `query_builders.rs` is a better move to `binders/builders/select_builder.rs`

**Review by BohuTANG — APPROVED** [review]

**Pull request merged**